### PR TITLE
IDEA vertex and silicon wrapper for IDEA_o1_v03

### PR DESCRIPTION
**Pull request by armin-ilg**
BEGINRELEASENOTES
- Update of IDEA vertex, with the ability to use the ultra-light vertex concept in-situ.
- No overlaps in all of vertex and silicon wrapper (not including the DDCAD imported vertex support and cooling cones yet), more performant silicon wrapper (only silicon wrapper barrel sensors are simplified)

ENDRELEASENOTES

Hi @BrieucF 

Here finally is the vertex and silicon wrapper, as presented in the FCC Week. This version is very performant, even with the sensitive surface module running. I push it directly to IDEA_o1_v03, so we can close the old [PR](https://github.com/key4hep/k4geo/pull/314). 
- This version also works with iLCSoft reconstruction (I had some issues with the old version. Apparently all barrel layers must be in the same envelope and this envelope has to be continuous). The "[CLD with IDEA vertex](https://github.com/armin-ilg/lcgeo/tree/CLD_with_IDEAvertex_and_ultra-light/FCCee/CLD_IDEAvertex/compact/CLD_o2_v05_IDEAvertex)" version, that I used for impact parameter resolution studies, is not included here however.
- I currently have the .obj files (other CAD format) directly in the repo. I see that for the MDI they are in an external place - shall I do the same?
- I'll work on the updated Silicon Wrapper design (as presented in FCC Week) soon - but this version here is okay for the moment I think. I'll make another MR for that.

Is there something you would like to see changed? I'm doing some final tests to see whether iLCSoft reco really is still working.

Cheers,
Armin

### CODE REVIEW

**Comment by atolosadelgado** [comment]
Hi,

@armin-ilg thanks a lot for the PR. Is it possible that the README in detector/tracker needs to be updated?
https://github.com/key4hep/k4geo/blob/main/detector/tracker/README.md

@BrieucF : I have checked the following:
```
* visualization in Geant4....... not possible, depth 2 already too complex for Geant4
* overlap vx, 300k points....... ok
* overlap IDEA (recursion 2).... ok
* geantino run.................. 30s/100k
* proton 10GeV, cos(theta), run. 652s/100k
```

Shall we merge the PR which enables downloading the beampipe CAD files as cmake configuration option, and then add the .obj in a separate PR, or do you want to add CAD models for the beampipe and Vertex in one single PR?

**Comment by BrieucF** [comment]
Hi Armin, Alvaro,

Thanks to both of you. Yes I propose to use the same approach as for MDI. And indeed we can do this in two separated PR's (merge the one for CAD and then tackle vertex). Since the support structure is now commented out, we can also proceed with this PR once the test are fixed. I am off today and tomorrow but please proceed without me.

File: install.sh
Comment (BrieucF) [inline install.sh:1]: Is this file intended to be here?

**Comment by BrieucF** [comment]
Thanks Armin! Can you update the ALLEGRO detector which is taking the vertex from IDEA?

You just have to change this symbolic link `FCCee/ALLEGRO/compact/ALLEGRO_o1_v03/Vertex_IDEA_o1_v01.xml` to point to `FCCee/IDEA/compact/IDEA_o1_v03/VertexComplete_IDEA_o1_v03.xml`

**Comment by BrieucF** [comment]
And update the branch

**Comment by armin-ilg** [comment]
Hi @BrieucF @atolosadelgado 

All done. Some final questions came to my mind:
- Does ALLEGRO foresee the RICH detector for PID or should I also here add the silicon wrapper?
- Should the lumical maybe also go into the MDI folder, as it is common to all concepts? (I e.g. moved the visibilities around for the IDEA lumical, but that would not apply to the others)

Anything else to be changed?

**Comment by armin-ilg** [comment]
About the [failed test](https://github.com/key4hep/k4geo/actions/runs/10389179975/job/28766599287?pr=363): Shall I revert the changes to the LumiCal or change all the LumiCal files in the same way?

**Comment by BrieucF** [comment]
Hi Armin,

> Does ALLEGRO foresee the RICH detector for PID or should I also here add the silicon wrapper?

Please add also the silicon wrapper to ALLEGRO ;-)

> Should the lumical maybe also go into the MDI folder, as it is common to all concepts? (I e.g. moved the visibilities around for the IDEA lumical, but that would not apply to the others)

Yes, it will go in the MDI folder, but this can be done in another PR. 

> About the [failed test](https://github.com/key4hep/k4geo/actions/runs/10389179975/job/28766599287?pr=363): Shall I revert the changes to the LumiCal or change all the LumiCal files in the same way?

This is a new test enforcing that files with the same name have exactly the same content. To pass the test you would just have to rename the xml file into e.g. `LumiCal_o1_v02.xml`. But the change seems very small and will not affect the physics so you could simply discard them from the PR. Can you show how the new parameters improve visualization?

**Comment by armin-ilg** [comment]
Addressed all of the above. About the visibilities: I had only changed where the "vis=..." were defined, nothing else.

I've now also added "GlobalTrackerReadoutID_DCH" to ALLEGRO, in the same way as it is for IDEA, I hope this is okay.

File: FCCee/ALLEGRO/compact/ALLEGRO_o1_v03/DectDimensions.xml [hunk 1]
```diff
@@ -18,36 +18,39 @@
     <constant name="world_side" value="6100*mm"/>
     <constant name="CrossingAngle" value="0.030*rad"/>  
 
-    <constant name="GlobalTrackerReadoutID" type="string" value="system:5,layer:3,stave:6,module:5,sensor:2"/>
-    <constant name="GlobalTrackerReadoutID_OB" type="string" value="system:5,layer:1,stave:6,module:4,sensor:2"/>
-    <constant name="GlobalTrackerReadoutID_Disk" type="string" value="system:5,side:-2,layer:3,petal:3,stave:6,module:10,sensor:2"/>
-    <constant name="GlobalTrackerReadoutID_SiWrapperB" type="string" value="system:5,layer:1,stave:9,module:6,sensor:1"/>
-    <constant name="GlobalTrackerReadoutID_SiWrapperD" type="string" value="system:5,side:1,layer:1,module:6,sensor:1"/>
+    <constant name="GlobalTrackerReadoutID_DCH" type="string" value="system:1,layer:16,phi:16"/>
```
Comment (andresailer) [inline FCCee/ALLEGRO/compact/ALLEGRO_o1_v03/DectDimensions.xml:21]: The whole point of the "Global" tracker readout ID is having a global one for all tracker readouts.

Comment (armin-ilg) [reply FCCee/ALLEGRO/compact/ALLEGRO_o1_v03/DectDimensions.xml:21]: My mistake for adding it here, apparently it is defined directly in the DriftChamber XML:
https://github.com/key4hep/k4geo/blob/bfe18998e04bafb3ccdf443f236c2b1b460f44c7/FCCee/IDEA/compact/IDEA_o1_v03/DriftChamber_o1_v02.xml#L127

**Comment by armin-ilg** [comment]
Fixed a last issue with ALLEGRO, now everything is final. No overlaps detected. Ready to merge @atolosadelgado @BrieucF @andresailer

**Comment by BrieucF** [comment]
Hi Armin, thanks! Two tests fail because of the renaming of the materials file under IDEA. Can you update those lines:
- https://github.com/key4hep/k4geo/blob/main/test/compact/DCH_standalone_o1_v02.xml#L17
- https://github.com/key4hep/k4geo/blob/main/test/compact/IDEA_withDRC_o1_v03.xml#L23

**Comment by BrieucF** [comment]
> No overlaps detected

Did you also check ALLEGRO?

**Comment by BrieucF** [comment]
Ha wait, sorry, I had not seen that the loading of the .stl files was commented out. This can therefore be handled in another PR. You can just remove the .stl from the PR then (and we should merge with "squash" to avoid having themin the commit history).

**Comment by armin-ilg** [comment]
> Hi Armin, thanks! Two tests fail because of the renaming of the materials file under IDEA. Can you update those lines:
> 
> * https://github.com/key4hep/k4geo/blob/main/test/compact/DCH_standalone_o1_v02.xml#L17
> * https://github.com/key4hep/k4geo/blob/main/test/compact/IDEA_withDRC_o1_v03.xml#L23

Done, I also updated the vertex and silicon wrapper in those tests, hope that's okay

**Comment by armin-ilg** [comment]
> Another important thing: similarly to the CAD beampipe, the binary stl files are too big for a gitHub repository (this PR would increase the repo size by 70 MB). I propose to use the same trick as for the beampipe: download the files at build time with this command: https://github.com/key4hep/k4geo/blob/main/CMakeLists.txt#L224 . I don't think you need to place this inside an `if` statement since by default those files are used. But please, do check whether the file are there already (https://github.com/key4hep/k4geo/blob/main/CMakeLists.txt#L220) and download them only if they are not.
> 
> I have placed the stl files here: https://fccsw.web.cern.ch/fccsw/filesForSimDigiReco/IDEA/IDEA_o1_v03/STL_FILES/
> 
> Thanks!

Done, will later make another MR with the way you propose

File: test/compact/IDEA_withDRC_o1_v03.xml [hunk 2]
```diff
@@ -43,13 +43,13 @@
...
   <!-- Import vertex -->
-  <include ref="../../FCCee/IDEA/compact/IDEA_o1_v03/Vertex_IDEA_o1_v01.xml"/>
+  <include ref="../../FCCee/IDEA/compact/IDEA_o1_v03/VertexComplete_IDEA_o1_v03.xml"/>
 
   <!-- Import drift chamber -->
   <include ref="../../FCCee/IDEA/compact/IDEA_o1_v03/DriftChamber_o1_v02.xml"/>
 
   <!-- Import silicon wrapper -->
-  <!-- <include ref="SiliconWrapper_o1_v01.xml"/> -->
+  <include ref="SiliconWrapper_o1_v03.xml"/>
```
Comment (BrieucF) [inline test/compact/IDEA_withDRC_o1_v03.xml]: You are missing the relative path here, this test still fails because of that. By the way, you can try the test before hand with `ctest` from the build directory.

Comment (armin-ilg) [reply test/compact/IDEA_withDRC_o1_v03.xml]: Yeah, just found that. Sorry, next time I'll first try it locally

**Comment by BrieucF** [comment]
Any further comment @andresailer ?

**Comment by andresailer** [comment]
> Any further comment @andresailer ?

No, all good from my side.

**Comment by BrieucF** [comment]
Wonderful! Thanks!

So, IDEA only misses a presampler now !! :-)

**Comment by armin-ilg** [comment]
Thanks for all the help, feedback and support!

**Comment by giovannimarchiori** [comment]
Great @armin-ilg! Can you please close the PR superseded by this one?

**Comment by armin-ilg** [comment]
> Great @armin-ilg! Can you please close the PR superseded by this one?

Thanks for reminding me. Done
